packages/compile-solidity/src/compilerSupplier/loadingStrategies/VersionRange.ts

```diff
@@ -180,7 +180,18 @@ export class VersionRange {
       throw error;
     }
     events.emit("downloadCompiler:succeed");
-    await this.cache.add(response.data, fileName);
+    try {
+      await this.cache.add(response.data, fileName);
+    } catch (error) {
+      if (error.message.includes("EACCES: permission denied")) {
+        const warningMessage =
+          "There was an error attempting to save the compiler to disk. " +
+          "The current user likely does not have sufficient permissions to " +
+          "write to disk in Truffle's compiler cache directory. See the error" +
+          `printed below for more information about this directory.\n${error}`;
```

Comment:
My first thought is "I don't think so". We don't really have a thorough enough understanding of the issue to really give explicit instructions on this so a message like that might just leave people scratching their head. I mean all we can tell them is try installing Node and npm with nvm; or maybe manually update the permissions to that folder.

Reply:
As far as putting the `try/catch` inside the method in `Cache`, I think I like what is in this PR better. What would doing that accomplish besides making this class a little bit less flexible. As I have it here it seems like we can use this and then control what happens in the case of a `throw`. I don't know if we might want to do different things in each case where it throws. I don't know, do you have thoughts?

Reply:
I think moving specific exception handling to `Cache` makes `VersionRange` more flexible. `EACCES` filtering is a Cache concern. If `Cache.add`'s implementation is w/in a `try-catch` it could consume `EACCES` exceptions and re-throw other catastrophic exceptions to its callers. The current implementation means, if we decide to use `Cache.add` elsewhere, we'd have to copy the exception handling and filtering as well. That being said, I'm not concerned with the current implementation. It's your call.

Reply:
I disagree, I think it is less flexible because we may not want to ignore these failures in another place where that method is implemented. Each usage might require different behavior. Warning inside cache would prevent this. 

However, maybe a solution to that problem would be to report a failure back to the caller. But then it just seems like that is the same as just letting it throw inside and handling it outside.

Reply:
This is `Cache.add`'s only client, and I'm content with the current implementation. Let's call this resolved.